Sources/Host/Darwin/Platform.cpp

```diff
@@ -118,6 +118,26 @@ std::string Platform::GetThreadName(ProcessId pid, ThreadId tid) {
 const char *Platform::GetSelfExecutablePath() {
   return LibProc::GetExecutablePath(getpid());
 }
+
+ErrorCode Platform::TranslateKernError(kern_return_t kret) {
+  switch (kret) {
+  case KERN_SUCCESS:
+    return ds2::kSuccess;
+  case KERN_FAILURE:
+    return ds2::kErrorUnknown;
+  case KERN_MEMORY_FAILURE:
+  case KERN_INVALID_ADDRESS:
+    return ds2::kErrorInvalidAddress;
+  case KERN_PROTECTION_FAILURE:
+  case KERN_NO_ACCESS:
+    return ds2::kErrorNoPermission;
+  case KERN_MEMORY_ERROR:
+  case KERN_RESOURCE_SHORTAGE:
+    return ds2::kErrorNoMemory;
+  default:
+    DS2BUG("unknown error code: %s [%d]", mach_error_string(kret), kret);
```

Comment:
nit: can you do `"unknown error code: %d [%s]"` instead (and swap the two following arguments). In other similar `DS2BUG`s we only print the numeric value and I'll add the string version that the system returns to match the code you just wrote here.